keps/sig-architecture/4330-compatibility-versions/README.md

```diff
@@ -610,21 +623,25 @@ The storage version of each group-resource is the newest
 
 ### API availability
 
-Similar to feature flags, all APIs group-versions declarations will be modified
-to track which Kubernetes version the API group-versions are introduced (or
-removed) at.
+Ideally, similar to feature flags, all APIs group-version-resource declarations should be modified
+to track which Kubernetes version the GVRs are introduced (or
+removed) at: if an API is introduced after (or removed before) the emulation version, it should not be available at the emulation version. But in practice, that would make code changes of API graduation that wants to use newer APIs intractable. 
+For example, if a controller is switching from Beta to GA API at the binary version, a full emulation would mean the whole controller code needs to be
+duplicated to select from the Beta or GA API based on the emulation version every time the API is used. That means a lot of extra work, complicated testing rules, and high maintenance cost even for simple API graduations.
 
-GA APIs should match the exact set of APIs enabled in GA for the Kubernetes version
-the emulation version is set to.
+Instead, we are proposing to keep forward compatibility of all APIs in compatibility version mode: 
+All API GVR declarations would be modified
+to track which Kubernetes version the API GVRs are removed at. APIs introduced between the emulation version and 
+the binary version are still available if the stability level is no less than the APIs of the same group and resource type introduced before or at the emulation version and enabled at the emulation version.
 
-All Beta APIs (both off-by-default and on-by-default, if any of those
-still exist) need to behave exactly as they did for the Kubernetes version
-the emulation version is set to. I.e. `--runtime-config=api/<version>` needs
-to be able to turn on APIs exactly like it did for each Kubernetes version that
-emulation version can be set to.
+If an API has GAed and has not been removed at the emulated version, it would be enabled by default at the emulation version. If a newer stable version of the GA API has been introduced between the emulation version and the binary version, the new GA API would also be enabled at the emulation version along with the old GA API.
```

Comment:
A comparison (maybe in a table), or how emulation works for normal APIs vs how it works for un-emulatable (non-emulatable?) features might make this information easier to digest.